src/traces/scatter/defaults.js

```diff
@@ -37,6 +37,7 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     var defaultMode = !stackGroupOpts && (len < constants.PTS_LINESONLY) ?
         'lines+markers' : 'lines';
     coerce('text');
+    coerce('texttemplate');
```

Comment:
So here, maybe we shouldn't coerce `text` when `texttemplate` is set? What do you think @antoinerg ?

Reply:
Very good question! Because `text` is an available flag in `textinfo`, I decided to make it available in `texttemplate` but that does not have to be the case.

Reply:
> So here, maybe we shouldn't coerce `text` when `texttemplate` is set?

If we remove `%{text}` from `texttemplate`, we still need to coerce it if `hovertemplate` is set.

Reply:
Hmm, very good point. 

I guess it would kinda of a mess to have logic for: 

_Do not coerce `text` is `texttemplate` is set OR `hovertemplate` is set_

---

So, we should coerce `text` always, but make it clear in the `texttemplate` attribute description that if `text` and `texttemplate` are set, `texttemplate` is used for on-graph text and `text` entries are available via `%{text}`.

Reply:
> So, we should coerce text always, but make it clear in the texttemplate attribute description that if text and texttemplate are set, texttemplate is used for on-graph text and text entries are available via %{text}.

I like this.

Reply:
Cool. Let's go with that!